File: atcorder/abc136/d/main.cpp
#include <bits/stdc++.h>
using namespace std;

#define ALL(v) v.begin(), v.end()
#define V vector
#define P pair
#define ld long double
#define ll long long
#define mod 1000000007
#define IINF INT_MAX
#define INF 1LL << 30


int main() {
    string s; cin >> s;

    int n = s.size();
    V<int> ans(n);

    for(int i = 0; i < 2; i++){
        int cnt = 0;
        for(int j = 0; j < n; j++){
            if(s[j] == 'R') cnt++;
            else{
                // 右は切り捨て
                ans[j] += cnt / 2;
                // 左は切り上げ
                ans[j - 1] += (cnt + 1) / 2;
                cnt = 0;
            }
        }
        // 反転してLをRに変えて同じ計算使い回す
        reverse(ALL(ans));
        reverse(ALL(s));
        for(int j = 0; j < n; j++){
            if(s[j] == 'L') s[j] = 'R';
            else s[j] = 'L';
        }
    }

    for(int i = 0; i < n; i++){
        cout << ans[i] << endl;
    }

    return 0;
}

File: atcorder/abc136/d/main.py
import sys
input = sys.stdin.readline
sys.setrecursionlimit(10**7)
inf = 10**17
mod = 10**9+7

s = input().rstrip()

n = len(s)
ans = [0] * n

rn = ln = 0
for i in range(n):
    if s[i] == 'R': rn += 1
    if s[i] == 'L': ln += 1
    if s[i] == 'L' and (i + 1 < n and s[i + 1] == 'R' or i == n - 1):
        ans[i - ln + 1] = ln - (ln // 2) + (rn // 2)
        ans[i - ln] = rn - (rn // 2) + (ln // 2)
        rn = ln = 0

for i in ans:
    print(i)
